docs/configure-artifact-repository.md

```diff
@@ -130,6 +140,34 @@ specify a bucket region in your workflow configuration.
     ...
 ```
 
+### AWS S3 with IAM credentials or S3 access grants
+
+Below is an example of how to use credentials from an [S3 access grant](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-grants.html)
+or temporary credentials from an IAM role. If you are using an IAM user, omit the `sessionTokenSecret` field.
+
+```yaml
+artifacts:
+  - name: my-output-artifact
+    path: /my-output-artifact
+    s3:
+      endpoint: s3.amazonaws.com
+      bucket: my-s3-bucket
+      # NOTE that, by default, all output artifacts are automatically tarred and
+      # gzipped before saving. So as a best practice, .tgz or .tar.gz
+      # should be incorporated into the key name so the resulting file
+      # has an accurate file extension.
```

Comment:
```suggestion
```

This isn't needed in this section (and it should probably be removed from this page entirely, as it is more specific to the artifacts page rather than configuring artifact repositories)